go-app.js

```diff
@@ -85,6 +87,8 @@ go.app = function() {
         question: question,
         check: function(content) {
           if (!go.utils.is_valid_msisdn(content)) {
+            return null; // vumi expects null or undefined if check passes
+          } else {
             return error;
```

Comment:
If it's not a valid_msisdn you want to return the error, the `return null;` and the `else` block can go, just and `if` and then return the `error`.